src/voting-tools-rs/src/testing/test_api/fake.rs

```diff
@@ -63,7 +63,19 @@ impl DataProvider for MockDbProvider {
                             if let Ok(data) = metadata.as_bytes() {
                                 Delegations::Legacy(hex::encode(data))
                             } else {
-                                Delegations::Delegated(vec![])
+                                let mut delegations = vec![];
+                                let delgation_list = metadata.as_list().unwrap();
+                                for i in 0..delgation_list.len() {
```

Comment:
Could we use `.enumerate()` here?

Reply:
I would love to but looks like MetadataList doesn't expose `.iter()`